modules/litemutes.ts

```diff
@@ -0,0 +1,207 @@
+  } else {
+    returnMessage = `That role does not exist, dummy.`
+  }
+  return message.channel.send(returnMessage)
+}
+
+const LiteMute = async (message: Message, args: string[]) => {
+  const LiteMuteTargetId = message.mentions.users.first().id
+  const curDate = new Date()
+  const LiteMutePeriod = args[1]
+    ? new Date(curDate.setSeconds(curDate.getSeconds() + parseInt(args[1], 10)))
+    : INDEFINITE_LiteMute
+  const LiteMuteRole: Role = await _getGuildLiteMutesRole(message.guild)
+  const LiteMuteTarget: GuildMember = await message.guild.fetchMember(LiteMuteTargetId)
+
+  try {
+    await LiteMuteTarget.addRole(LiteMuteRole)
+  } catch {
+    return message.channel.send("I don't have permission to manage roles!")
+  }
+  await message.client['timerData'].set(`${message.guild.id}.${LiteMuteTargetId}`, LiteMutePeriod)
+
+  const LiteMutePeriodText = LiteMutePeriod === INDEFINITE_LiteMute ? 'indefinitely' : `for ${args[1]} seconds`
+  return message.channel.send(`User ${args[0]} has been LiteMuted ${LiteMutePeriodText}.`)
+}
+
+const unLiteMute = async (message: Message, args: string[]) => {
+  const LiteMuteTargetId = message.mentions.users.first().id
+  const LiteMuteRole = await _getGuildLiteMutesRole(message.guild)
+  const LiteMuteTarget = await message.guild.fetchMember(LiteMuteTargetId)
+
+  try {
+    await LiteMuteTarget.removeRole(LiteMuteRole)
+  } catch {
+    return message.channel.send("I don't have permission to manage roles!")
+  }
+
+  message.client['timerData'].delete(`${message.guild.id}.${LiteMuteTargetId}`)
+  return message.channel.send(`User ${args[0]} has been unLiteMuted.`)
+}
+
+const newcomerLiteMuteCheck = async (member: GuildMember) => {
+  const LiteMuteRole = await _getGuildLiteMutesRole(member.guild)
+  if (member.client['timerData'].has(`${member.guild.id}.${member.id}`)) {
+    try {
+      await member.addRole(LiteMuteRole)
+    } catch (e) {
+      log.warn(`${name}.newcomerLiteMuteCheck: ${e}`)
+      return
+    }
+  }
+}
+
+const processManualLiteMute = async (previous: GuildMember, actual: GuildMember) => {
+  let LiteMuteRole: Role | null
+
+  try {
+    LiteMuteRole = await _getGuildLiteMutesRole(actual.guild)
+  } catch (e) {
+    log.warn(e)
+    return
+  }
+
+  const isListed = actual.client['timerData'].has(`${actual.guild.id}.${actual.id}`)
+  const hadRole = previous.roles.has(LiteMuteRole.toString())
+  const hasRole = actual.roles.has(LiteMuteRole.toString())
+
+  if (!isListed && !hadRole && hasRole) {
+    await actual.client['timerData'].set(`${actual.guild.id}.${actual.id}`, INDEFINITE_LiteMute)
+  }
+  if (isListed && hadRole && !hasRole) {
+    await actual.client['timerData'].delete(`${actual.guild.id}.${actual.id}`)
+  }
+}
+
+const removeLiteMuteFromBannedUser = async (guild: Guild, user: User) => {
+  return _removeMemberFromLiteMutes(guild.client, `${guild.id}.${user.id}`)
+}
+
+export const name = 'LiteMute'
```

Comment:
The issue specifies the litemutes module should be optional, given that some guilds may not have one. Please correct the module specification as to not load it on boot